### PR TITLE
Fix mutiple api calls for user urls api

### LINKED ISSUE #129: Mutiple api calls for user urls
### Issue Description
User URLs API is called multiple times when we visit the dashboard page of the user

### Expected Behavior
API should call when there are state changes or the page is refreshed

### Screenshots

https://github.com/user-attachments/assets/ba5c2d25-ed48-4622-adee-17215046c3ef

### Reproducibility

 - [ ] This issue is reproducible
 - [ ] This issue is not reproducible

### Steps to Reproduce
1. 
2. 

### Severity/Priority
 - [ ] Critical
 - [ ] High
 - [ ] Medium
 - [ ] Low

### Checklist

- [ ] I have read and followed the project's code of conduct.
- [ ] I have searched for similar issues before creating this one.
- [ ] I have provided all the necessary information to understand and reproduce the issue.
- [ ] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by vinit717**
## Issue Ticket Number

-  closes #129 

## Description

Fix multiple calls of user URLs APi

**Breaking Changes**

-   [ ] Yes
-   [x] No

_If your feature introduces breaking changes or if something is missing, please mention the related issue tickets._

**Development Tested?**

-   [x] Yes
-   [ ] No

_Confirm whether the changes have been tested locally during development._

**Tested in Staging?**

-   [x] Yes
-   [ ] No

_Indicate whether the changes have been tested in the staging environment._

**Under Feature Flag**

-   [x] Yes
-   [ ] No

_Specify if the changes are currently under a feature flag._

**Database Changes**

-   [ ] Yes
-   [x] No

_Indicate whether the changes include modifications to the database._

### Screenshots

https://github.com/user-attachments/assets/74ca5400-56f9-4310-9c1c-da574abe55c4

## Test Coverage
image
## Additional Notes

Include any additional notes, considerations, or explanations that might be helpful for reviewers.

### CODE REVIEW

File: src/services/api.tsx [hunk 1]
```diff
@@ -56,6 +56,8 @@ const useGetUrlsQuery = ({ enabled = true }: { enabled?: boolean }) => {
     return useQuery({
         queryKey: ['urls'],
         enabled: enabled,
+        retry: false,
```
Comment (prakashchoudhary07) [inline src/services/api.tsx:59]: When does it retry?

Comment (iamitprakash) [reply src/services/api.tsx:59]: @vinit717

Comment (vinit717) [reply src/services/api.tsx:59]: Sorry I just remove this 
As retry is basically to again and again retry the query till the no we set before showing any error

File: src/services/api.tsx [hunk 2]
```diff
@@ -56,6 +56,7 @@ const useGetUrlsQuery = ({ enabled = true }: { enabled?: boolean }) => {
     return useQuery({
         queryKey: ['urls'],
         enabled: enabled,
+        refetchOnWindowFocus: false,
```
Comment (yesyash) [inline src/services/api.tsx:59]: - can we move this to the global config (queryClient in _app.tsx) so that other get APIs have the same functionality?

Comment (vinit717) [reply src/services/api.tsx]: Done

**Review by yesyash — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]

**Review by iAmAshuSahoo — APPROVED** [review]

**Pull request merged**